src/cryptographic_primitives/commitments/pedersen_commitment.rs

```diff
@@ -0,0 +1,52 @@
+/*
+    Cryptography utilities
+
+    Copyright 2018 by Kzen Networks
+
+    This file is part of Cryptography utilities library
+    (https://github.com/KZen-networks/cryptography-utils)
+
+    Cryptography utilities is free software: you can redistribute
+    it and/or modify it under the terms of the GNU General Public
+    License as published by the Free Software Foundation, either
+    version 3 of the License, or (at your option) any later version.
+
+    @license GPL-3.0+ <https://github.com/KZen-networks/cryptography-utils/blob/master/LICENSE>
+*/
+
+use super::traits::Commitment;
+use elliptic::curves::secp256_k1::Secp256k1Point;
+use elliptic::curves::secp256_k1::Secp256k1Scalar;
+use elliptic::curves::traits::*;
+
+use BigInt;
+
+pub struct PedersenCommitment;
+
+impl Commitment<Secp256k1Point> for PedersenCommitment {
```

Comment:
yes, the reason was that not every elliptic curve that is used for signing can be used for Pedersen commitment as well so when new elliptic curves are introduced we would add pedersen commitment based on them. But your point is solid. @gbenattar @romanz  do you know of a way to support Pederesen with fixed elliptic curve while condition compilation on all other elliptic curve related stuff? this might come in useful in the future?

Reply:
out of curiosity, what determines whether a curve can be used for commitment or not?

for conditional compilation you can simply annotate the impl with `#[cfg(feature = "...")]`

Reply:
If the curve is prime order group or not. For example curve25519 is not.

Reply:
I sprayed `#[cfg(feature = "curvesecp256k1")]` all over the place. It appears that we are using pedersen commitments in many places. @romanz @mortendahl  @gbenattar maybe one of you have an idea on how to do this exclusion more elegant or at least how to minimize the times I use `#[cfg(feature = "curvesecp256k1")]`?